Paintroid/src/main/java/org/catrobat/paintroid/presenter/MainActivityPresenter.java

```diff
@@ -239,6 +272,46 @@ public void handleActivityResult(@ActivityRequestCode int requestCode, int resul
 		}
 	}
 
+	@Override
+	public void handlePermissionRequestResults(int requestCode, String[] permissions, int[] grantResults) {
+		switch (requestCode) {
```

Comment:
Please simplify.
If the permission is not granted you can call
```
navigator.showPermissionDialog(PermissionInfoDialog.PermissionType.EXTERNAL_STORAGE, EXTERNAL_STORAGE_PERMISSION_DIALOG, requestCode);
```
If the permission is granted, switch over the cases and call the corresponding functions.